rust/src/mime/smtp_log.rs

```diff
@@ -197,21 +197,22 @@ pub unsafe extern "C" fn SCMimeSmtpLogFieldString(
 }
 
 fn log_data_header(
-    js: &mut JsonBuilder, ctx: &MimeStateSMTP, hname: &str,
+    js: &mut JsonBuilder, ctx: &MimeStateSMTP, hname: &str, field: &str,
 ) -> Result<(), JsonError> {
     for h in &ctx.headers[..ctx.main_headers_nb] {
         if mime::slice_equals_lowercase(&h.name, hname.as_bytes()) {
-            js.set_string(hname, &String::from_utf8_lossy(&h.value))?;
+            js.set_string(field, &String::from_utf8_lossy(&h.value))?;
             break;
         }
     }
     return Ok(());
 }
```

Comment:
This same function is being changed in https://github.com/OISF/suricata/pull/12904, so maybe we wait for that one and then rebase on top of that?

Or have both works as a single PR? Dunno what's preferred.